src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  combineLatest,
+  EMPTY,
+  Observable,
+  of as observableOf,
+  Subject,
+  Subscription,
+} from "rxjs";
+import {
+  catchError,
+  distinctUntilChanged,
+  map,
+  mergeMap,
+  shareReplay,
+  take,
+} from "rxjs/operators";
+import { QueuedSourceBuffer } from "../../../core/source_buffers";
+import { Representation } from "../../../manifest";
+import arrayFindIndex from "../../../utils/array_find_index";
+import castToObservable from "../../../utils/cast_to_observable";
+import concatMapLatest from "../../../utils/concat_map_latest";
+import PPromise from "../../../utils/promise";
+import {
+  areRangesOverlapping,
+  convertToRanges,
+} from "../../../utils/ranges";
+import request from "../../../utils/request";
+import appendInitSegment from "./append_init_segment";
+import prepareSourceBuffer from "./media_source";
+
+interface IThumbnailInfo {
+  start: number;
+  duration: number;
+  mediaURL: string;
+}
+
+interface IThumbnailTrack {
+  thumbnailInfos: IThumbnailInfo[];
+  initURL: string;
+  codec: string;
+}
+
+const MAXIMUM_MEDIA_BUFFERED = 2;
+
+/**
+ * Get segment data from cache or load it.
```

Comment:
Which cache are you speaking of?
This comment make it like it's a cache we maintain